src/components/camera/server.hpp

```diff
@@ -27,7 +27,7 @@ class CameraServer : public ResourceServerBase,
         const ::viam::component::camera::v1::GetPropertiesRequest* request,
         ::viam::component::camera::v1::GetPropertiesResponse* response) override;
 
-    void register_server() override;
+    void register_server(Server* server) override;
```

Comment:
For consistency's sake, can we use a smart pointer to `server` here and in generic?

Reply:
Switched to `std::shared_ptr<Server>` everywhere.